wpilib/wpilib/timedrobot.py

```diff
@@ -0,0 +1,52 @@
+# validated: 2017-11-09 TW ef3267833fc3 edu/wpi/first/wpilibj/TimedRobot.java
+# ----------------------------------------------------------------------------
+# Copyright (c) FIRST 2008-2012. All Rights Reserved.
+# Open Source Software - may be modified and shared by FRC teams. The code
+# must be accompanied by the FIRST BSD license file in the root directory of
+# the project.
+# ----------------------------------------------------------------------------
+
+import hal
+
+from .iterativerobotbase import IterativeRobotBase
+from .notifier import Notifier
+from .timer import Timer
+
+
+__all__ = ["TimedRobot"]
+
+
+class TimedRobot(IterativeRobotBase):
+    DEFAULT_PERIOD = .02
+
+
+    def __init__(self):
+        super().__init__()
+        hal.report(hal.UsageReporting.kResourceType_Framework, hal.UsageReporting.kFramework_Iterative)
+
+        self.period = TimedRobot.DEFAULT_PERIOD
+        self.startLoop = False
+        self.loop = Notifier(self.loopFunc)
+
+    def startCompetition(self):
+        self.robotInit()
+
+        hal.observeUserProgramStarting()
+
+        self.startLoop = True
+        self.loop.startPeriodic(self.period)
+
+        while True:
+            Timer.delay(60*60*24)
```

Comment:
Do things break if this were a `time.sleep()` call instead?

Reply:
No but I figured it better to use hal.sleep for...reasons?

Reply:
simulation?

Reply:
What do you mean?

Reply:
We should add it to the porting notes -- never use time.sleep, because simulation reasons.

Reply:
Wow I did something right for once

Reply:
what is this sleep doing, anyways? it looks like the intent is to keep the calling thread alive but not consuming cpu resources. would hal sleep vs system sleep really make any difference?

Reply:
I suspect tests would hang if you called `time.sleep()`. It'd be fine on the real robot though.